### PR TITLE
Disable Use Object Initializer diagnostic prior to C# 3

### PR DESCRIPTION
**Pull request by jbevain**

### CODE REVIEW

**Comment by Pilchie** [comment]
Thanks for the contribution @jbevain  - please sign the CLA :-)

@dotnet/roslyn-ide (especially @CyrusNajmabadi) please review.

**Comment by CyrusNajmabadi** [comment]
👍

File: src/Features/CSharp/Portable/UseObjectInitializer/CSharpUseObjectInitializerDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -21,6 +21,11 @@ internal class CSharpUseObjectInitializerDiagnosticAnalyzer :
     {
         protected override bool FadeOutOperatorToken => true;
 
+        // object initializers are only available in C# 3.0 and above.  Don't offer this refactoring
+        // in projects targeting a lesser version.
+        protected override bool AreObjectInitializersSupported(SyntaxNodeAnalysisContext context) =>
+            ((CSharpParseOptions) context.Node.SyntaxTree.Options).LanguageVersion >= LanguageVersion.CSharp3;
```
Comment (jbevain) [inline src/Features/CSharp/Portable/UseObjectInitializer/CSharpUseObjectInitializerDiagnosticAnalyzer.cs]: Fixed.

Comment (jasonmalinowski) [inline src/Features/CSharp/Portable/UseObjectInitializer/CSharpUseObjectInitializerDiagnosticAnalyzer.cs]: I'd also prefer the comments to be closer to the logic (for here and in the VB one too). That reads as more a generic doc comment.

File: src/Features/CSharp/Portable/UseObjectInitializer/CSharpUseObjectInitializerDiagnosticAnalyzer.cs [hunk 2]
```diff
@@ -21,6 +21,11 @@ internal class CSharpUseObjectInitializerDiagnosticAnalyzer :
     {
         protected override bool FadeOutOperatorToken => true;
 
+        // object initializers are only available in C# 3.0 and above.  Don't offer this refactoring
```
Comment (jbevain) [inline src/Features/CSharp/Portable/UseObjectInitializer/CSharpUseObjectInitializerDiagnosticAnalyzer.cs]: Fixed.

**Comment by CyrusNajmabadi** [comment]
Merged in.  Thanks for the contribution!

**Pull request merged**